### ISSUE TITLE
Promise does not reject/resolve when doing update with set({})

### ISSUE THREAD

**Issue by threema-danilo** [juanluispaz/ts-sql-query#27]
To reproduce, I modified the file at `./src/examples/BetterSqlite3SyncronousExample.ts`:

```diff
diff --git a/src/examples/BetterSqlite3SyncronousExample.ts b/src/examples/BetterSqlite3SyncronousExample.ts
index ea00711..f31c0d3 100644
--- a/src/examples/BetterSqlite3SyncronousExample.ts
+++ b/src/examples/BetterSqlite3SyncronousExample.ts
@@ -81,7 +81,7 @@ function main() {
             .returningLastInsertedId()
             .executeInsert())
         assertEquals(i, 1)
-
+
         i = sync(connection
             .insertInto(tCustomer)
             .values({ firstName: 'Other', lastName: 'Person', companyId: 1 })
@@ -175,6 +175,12 @@ function main() {
             .executeUpdate())
         assertEquals(i, 1)

+        sync(connection
+            .update(tCompany)
+            .set({})
+            .where(tCompany.id.equals(2))
+            .executeUpdate())
+
         name = sync(connection
             .selectFrom(tCompany)
             .where(tCompany.id.equals(2))
@@ -432,4 +438,4 @@ function sync<T>(promise: Promise<T>): T {
         throw error
     }
     return result
```

When running this, the following happens:

> Error: You performed a real async operation, not a database operation, inside the function dedicated to calling the database

I suspect the reason is that an empty `set({})` results in a non-resolving promise that doesn't error either.

This isn't a contrived scenario, because if you're doing this:

```typescript
.set({
  fieldA: optionalObject?.requiredField,
  fieldB: requiredObject.optionalField,
})
```

...it may happen that both are `undefined`, which results in the same issue.

**Comment by juanluispaz** [comment]
Ups, sorry for the inconvenience.

Fixed, now when there is no columns to set a proper promise (in this case, a synchronous one) is returned indicated no update were performed, as in the other implementations.

Published, among other changes, in [ts-sql-query 1.24.0](https://ts-sql-query.readthedocs.io/en/stable/CHANGELOG/#v1240-21-dec-2021)

**Comment by juanluispaz** [comment]
**Note**: 

the `.set({...})` method will set all the columns that exists in the provided object, even if the value is null or undefined (in this case null will set in the database). I suppose in your example the object provided to the set method is constructed outside and it ended with no properties. 

There is a `.setIfValue({...})` that will be more suitable for your example:

```
.setIfValue({
  fieldA: optionalObject?.requiredField,
  fieldB: requiredObject.optionalField,
})
```

In this way you ensure the field will be set only when there is a provided value. Any null, undefined or empty string (if you didn't disable this last one) will be ignored.

The `set` and `setIfValue` allows you to pick the proper strategy to deal with absent values, and, that depends on the business logic you want to implement. You can combine (use both of them) if you want.

Any case, if you execute an update where there is no columns to update, a promise indicating no values were updated will be returned (no query will be executed in the database because it will be an invalid query). This was the specific case detected and fixed in this bug.

**Comment by threema-danilo** [comment]
> Published, among other changes, in ts-sql-query 1.24.0

Thanks for the fast fix!

> the `.set({...})` method will set all the columns that exists in the provided object, even if the value is null or undefined (in this case null will set in the database). I suppose in your example the object provided to the set method is constructed outside and it ended with no properties.

I forgot to mention: We're using `set({...}).ignoreAnySetWithNoValue()`. I haven't looked at the details (the implementation is still rough), but that's the same as `setIfValue({...})`, right? Or is this meant to deal with `set(null)`?

**Comment by juanluispaz** [comment]
They are the same, but apply over different scope:

- `.ignoreAnySetWithNoValue()` filter all previous set and ignore all that have no value (null, undefined, empty string)
- `setIfValue({...})` only filter the set provided by argument, all previous set will remain the same.

You have the possibility to call `set` or any of its variants ( you can see [all variants here](https://ts-sql-query.readthedocs.io/en/stable/supported-operations/#update-definition) ) multiple times. This distinction allows you combine in different ways all the options to create complex update queries.

In few words:
`set({...}).ignoreAnySetWithNoValue()` is the same as `setIfValue({...})` when you have a single call to `set`.

Example of difference: 

```ts
.set({
   myColumn: predefinedValue
}.set({
   myColumn: providedOptionalValue
}).ignoreAnySetWithNoValue()
```

If `providedOptionalValue` have no value, the update will have no set. But, if you do:

```ts
.set({
   myColumn: predefinedValue
}.setIfValue({
   myColumn: providedOptionalValue
})
```

If `providedOptionalValue` have no value, the update will have `{ myColumn: predefinedValue }`.

Be aware `predefinedValue` can be a sql expression (like `connection.currentDateTime()`), meanwhile the second set can be an object coming as argument of the current function.
